### PR TITLE
Implement PatientConvergedEarlyStopping.

### PR DESCRIPTION
**Pull request by kaminow**
Implement new early stopping method (`patient_converged`) that checks for convergence and then continues with the training loop to ensure that the model has actually converged and not just reached a local minimum.

### CODE REVIEW

File: asapdiscovery-ml/asapdiscovery/ml/config.py [hunk 1]
```diff
@@ -175,17 +180,30 @@ def check_args(cls, values):
                     "Values required for n_check and divergence when using "
                     "ConvergedEarlyStopping."
                 )
+            case EarlyStoppingType.patient_converged:
+                assert (values["n_check"] is not None) and (
```
Comment (hmacdope) [inline asapdiscovery-ml/asapdiscovery/ml/config.py:184]: ValueError rather than assert.

**Review by hmacdope — APPROVED** [review]
LGTM, just one nitpick

**Comment by kaminow** [comment]
@hmacdope I'm pretty sure not, but is this failing test something I broke? or can we just push ahead?

**Comment by hmacdope** [comment]
Ill fix the test #910

**Pull request merged**